docs/source/developer/committer-notes.rst

```diff
@@ -0,0 +1,162 @@
+
+We need linear code history, without unanticipated changes introduced by merge commits. This is important so that contributions are validated using the exact code that will land on ``main``.
+
+We need separate release branches to allow bug fixes for previous versions. This is important to support library users who can't upgrade to the latest version of OpenDP yet.
+
+All release tasks should be automated and drivable from GitHub. Creating a release should not require setup of a local environment, and should not have any dependencies on non-standard tools. This is important to allow for delegation of tasks and continuity of the project.
+
+
+Task Tracking
+=============
+
+* Use GitHub Issues to track all tasks. This is helpful to know who's working on what.
+* Use the `OpenDP Development GitHub Project <https://github.com/orgs/opendp/projects/1?card_filter_query=label%3A%22opendp+core%22>`_ queue up tasks for work.
+* Manage all changes using GitHub Pull Requests.
+
+Code Hygiene
+============
+
+* Follow the Rust guidelines for coding style. Code should be formatted using the default settings of rustfmt. (Should we enforce this on PRs?)
+* Write API docs for all public and significant private APIs. We use inline comments to explain complicated code.
+* Make sure ``main`` is always in a good state: Code compiles, tests pass. If ``main`` is ever broken, it should be the team's top priority to fix it.
+* Use GitHub Actions to check PRs automatically, and don't allow merges if checks fail.
+
+Branching Strategy
+==================
+
+* Use a single, long-lived branch named ``main`` for core project history.
+* Maintain a linear history on ``main``, meaning every commit is based only on the previous commit. Don't do merge commits onto ``main``.
+* Do development on short-lived feature branches, derived from ``main``. Feature branches have the naming scheme ``<nnn>-<short-desc>``, where ``<nnn>`` is the number of the GitHub issue tracking this task, and ``<short-desc>`` is a short description of the change. For instance, ``123-new-measurement``
+* Manage all changes using GitHub PRs from feature branches onto main. Check for test success and do code reviews before approving PRs.
+* To maintain linear history, require PRs to be up to date with main. This means that developers may need to rebase feature branches periodically.
+* Try to keep PRs relatively small and self contained. This simplifies code reviews, and reduces the likelihood of rebasing hassles.
+* Generally, squash feature branches when merging PRs, so that there's a 1-to-1 correspondence between issues/PRs and commits.
+* To enforce this strategy, use the following branch protections on main:
+  * Require pull request reviews before merging
+  * Dismiss stale pull request approvals when new commits are pushed
+  * Require status checks to pass before merging
+  * Require branches to be up to date before merging
+  * Require linear history
```

Comment:
These sub bullets don't come through:

![Screen Shot 2021-08-30 at 4 15 07 PM](https://user-images.githubusercontent.com/21006/131399533-7919e3d6-1ee8-4eb4-b8bd-dbdf272ab143.png)

I think you have to add a newline before the first sub bullet.

Reply:
Yes, thanks for catching. I find this very annoying about `.rst` format!